packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -53,236 +60,283 @@ function findResponseByStatusCode(
+            headers: headers || [],
+          });
+        }
       }
-    }
-    // user did not provide mediaType
-    // OR
-    // a httpContent for a provided mediaType does not exist
-    return helpers.negotiateDefaultMediaType(
-      {
-        code,
-        dynamic,
-        exampleKey,
-      },
-      response,
-    );
+      // user did not provide mediaType
+      // OR
+      // a httpContent for a provided mediaType does not exist
+      logger.trace('No mediaType provided. Fallbacking to the default media type (application/json)');
+      return helpers.negotiateDefaultMediaType(
+        {
+          code,
+          dynamic,
+          exampleKey,
+        },
+        response,
+      );
+    });
   },
 
   negotiateOptionsForDefaultCode(
     httpOperation: IHttpOperation,
     desiredOptions: NegotiationOptions,
-  ): IHttpNegotiationResult {
+  ): Reader<Logger, Either<Error, IHttpNegotiationResult>> {
     const lowest2xxResponse = findLowest2xx(httpOperation.responses);
     if (lowest2xxResponse) {
       return helpers.negotiateOptionsBySpecificResponse(httpOperation, desiredOptions, lowest2xxResponse);
     }
-    throw new Error('No 2** response defined, cannot mock');
+
+    return reader.of(left(new Error('No 2** response defined, cannot mock')));
   },
 
   negotiateOptionsBySpecificCode(
     httpOperation: IHttpOperation,
     desiredOptions: NegotiationOptions,
     code: string,
-  ): IHttpNegotiationResult {
+  ): Reader<Logger, Either<Error, IHttpNegotiationResult>> {
     // find response by provided status code
-    const responseByForcedStatusCode =
-      findResponseByStatusCode(httpOperation.responses, code) ||
-      createResponseFromDefault(httpOperation.responses, code);
-    if (responseByForcedStatusCode) {
-      try {
-        // try to negotiate
-        return helpers.negotiateOptionsBySpecificResponse(httpOperation, desiredOptions, responseByForcedStatusCode);
-      } catch (error) {
-        // if negotiations fail try a default code
-        try {
-          return helpers.negotiateOptionsForDefaultCode(httpOperation, desiredOptions);
-        } catch (error2) {
-          throw new Error(`${error}. We tried default response, but we got ${error2}`);
-        }
+    return withLogger(logger => {
+      const result = findResponseByStatusCode(httpOperation.responses, code);
+      if (!result) {
+        logger.info(`Unable to find a ${code} response definition`);
+        return createResponseFromDefault(httpOperation.responses, code);
       }
-    }
-    // if no response found under a status code throw an error
-    throw new Error('Requested status code is not defined in the schema.');
+
+      return result;
+    }).chain(responseByForcedStatusCode => {
+      if (responseByForcedStatusCode) {
+        return helpers
+          .negotiateOptionsBySpecificResponse(httpOperation, desiredOptions, responseByForcedStatusCode)
+          .chain(either =>
+            either.fold(
```

Comment:
One of the goals was to improve readability but I can't tell this did the trick.
We're a increasing the cognitive complexity. Contributing to Prism is already hard (TypeScript, monorepo, workspaces/build and now we're adding functional programming on top of it - I have the feeling we're narrowing potential contributors to 0.00001% of developers ;) )

Reply:
There is nothing to do with this comment other than maybe not using `chains`?

Reply:
`chain` is a fundamental piece — I'll try to explain it in another of the comment you did.

You have a point though, this is indeed a case-limit that will be addressed soon by the library itself by providing a more specific interface to handle this use case. So you're not wrong here!